### PR TITLE
post: nv12-vkms

### LINKED ISSUE #3: Write a blogpost about how VKMS work is going

### PR DESCRIPTION
**Pull request by Grillo-0**
closes #3

### CODE REVIEW

**Comment by tales-aparecida** [comment]
Might be relevant as an additional reference: https://docs.kernel.org/userspace-api/media/v4l/pixfmt-yuv-planar.html

File: _posts/2023-06-03-nv12-vkms.md [hunk 1]
```diff
@@ -0,0 +1,508 @@
+---
+layout: post
+title: "How it's Going: Adding NV12 support to VKMS"
+date: 2023-06-03 12:30
+category: kernel developing
+tags: open-source graphics
+---
+
+This work proposed by my mentor [Maira Canal][maira] is going more difficult than I
```
Comment (tales-aparecida) [inline _posts/2023-06-03-nv12-vkms.md]:
```suggestion
This work proposed by my mentor [Maíra Canal][maira] is going more difficult than I
```

File: _posts/2023-06-03-nv12-vkms.md [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+ }
+```
+
+The conversion code was taken from the [`tpg-core.c`][tpg-core.c], a virtual
+driver from the media subsystem that does those conversions in software as well.
+
+As a side thought, maybe would be better to have those two subsystems use the
+same code, maybe with a separate subsystem that handles color formats.
+
+The TPG code was changed to use the [`drm_fixed.h`][drm_fixed] operations, as to
```
Comment (tales-aparecida) [inline _posts/2023-06-03-nv12-vkms.md]:
```suggestion
The TPG code was changed to use the [`drm_fixed.h`][drm_fixed] operations, to
```

File: _posts/2023-06-03-nv12-vkms.md [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+![1D LUT representation](/assets/img/1d-gamma-lut.webp)
+
+There is a more complex type of LUT, a 3D LUT. But this one I don't fully
+understand and it's not needed. All I know is that you use the three color
+channels at the same time for the index, one for each coordinate, and the value
+that you get is a color. And besides that, you have to do interpolations.
+
+![3D LUT representation](/assets/img/3d-gamma-lut.webp){: width="50%" }
+
+It image representation is a pretty cube :).
```
Comment (tales-aparecida) [inline _posts/2023-06-03-nv12-vkms.md]:
```suggestion
Its visual representation is a pretty cube :).

```

**Review by tales-aparecida — CHANGES_REQUESTED** [review]
I don't know much about this, let's wait for more reviewers

File: _posts/2023-06-03-nv12-vkms.md [hunk 4]
```diff
@@ -0,0 +1,508 @@
...
+---
+
+This work proposed by my mentor [Maíra Canal][maira] is going more difficult than I
+thought >W<.
+
+# What is VKMS?
+
+The [Virtual Kernel Mode Setting][vkms] is a virtual video driver. It is used for
+testing the correct display of userspace programs without needing a physical
+monitor. This technique is useful for CI/CD for example.
```
Comment (andrealmeid) [inline _posts/2023-06-03-nv12-vkms.md]:
```suggestion
monitor. Virtual drivers are useful for automated testing.
```

File: _posts/2023-06-03-nv12-vkms.md [hunk 5]
```diff
@@ -0,0 +1,508 @@
...
++                                         BIT(DRM_COLOR_YCBCR_LIMITED_RANGE) |
++                                         BIT(DRM_COLOR_YCBCR_FULL_RANGE),
++                                         DRM_COLOR_YCBCR_BT601,
++                                         DRM_COLOR_YCBCR_FULL_RANGE);
++
+        return plane;
+ }
+```
+
+The conversion code was taken from the [`tpg-core.c`][tpg-core.c], a virtual
```
Comment (andrealmeid) [inline _posts/2023-06-03-nv12-vkms.md]: this hyperlink seems broken

**Comment by andrealmeid** [comment]
- I think I would appreciate a small introduction about the post before jumping directly to "what's vkms, what's NV12"
- There's a public domain image that helps illustrating YUV https://en.wikipedia.org/wiki/YUV#/media/File:Barn-yuv.png

**Comment by andrealmeid** [comment]
@Grillo-0 LGTM!

**Comment by tales-aparecida** [comment]
if it LGT andrealmeid, it LGTM as well. ;D

**Review by tales-aparecida — APPROVED** [review]

**Pull request merged**